spec/requests/landing_pages_spec.rb

```diff
@@ -0,0 +1,245 @@
+
+  let!(:domain) { "market.custom.org" }
+
+  before do
+    @community = FactoryGirl.create(:community, :domain => domain, use_domain: true)
+    @community.reload
+
+    10.times do
+      FactoryGirl.create(:category, community_id: @community.id)
+    end
+
+    10.times do
+      FactoryGirl.create(:listing, community_id: @community.id)
+    end
+
+    sample_structure = JSON.parse(CustomLandingPage::ExampleData::TEMPLATE_STR)
+    sample_structure = populate_categories(sample_structure, @community.categories.map(&:id))
+    sample_structure = populate_listings(sample_structure, @community.listings.map(&:id))
+
+    create_version(@community.id, sample_structure, 1)
+    create_version(@community.id, sample_structure, 2)
+
+    FactoryGirl.create(
+      :landing_page,
+      community_id: @community.id,
+      enabled: 0,
+      released_version: nil
+    )
+  end
+
+  context "when not released" do
+    it "landing page is disabled" do
+      expect(::CustomLandingPage::LandingPageStore.enabled?(@community.id)).to eq false
+    end
+
+    it "index routes to homepage" do
+      expect_controller("http://#{domain}", "homepage", "index")
+    end
+
+    it "search path redirects to homepage" do
+      get "http://#{domain}/s"
+
+      expect(response.status).to eq(307)
+      expect(response.location).to eq("http://#{domain}/")
+    end
+
+    it "preview routes to landing page preview" do
+      expect_controller("http://#{domain}/_lp_preview?preview_version=1", "landing_page", "preview")
+    end
+
+    it "renders correct preview" do
+      expect_string("http://#{domain}/_lp_preview?preview_version=1", "<title>Title version 1</title>")
+      expect_string("http://#{domain}/_lp_preview?preview_version=2", "<title>Title version 2</title>")
+    end
+  end
+
+  context "when released" do
+    before do
+      release_version(@community.id, 1)
+    end
+
+    it "landing page is enabled" do
+      expect(::CustomLandingPage::LandingPageStore.enabled?(@community.id)).to eq true
+    end
+
+    it "index routes to landing page" do
+      expect_controller("http://#{domain}", "landing_page", "index")
+    end
+
+    it "search path routes to search" do
+      expect_controller("http://#{domain}/s", "homepage", "index")
+    end
+
+    it "shows correct landing page version" do
+      expect_string("http://#{domain}", "<title>Title version 1</title>")
+    end
+
+    it "preview routes to landing page preview" do
+      expect_controller("http://#{domain}/_lp_preview?preview_version=1", "landing_page", "preview")
+    end
+
+    context "new version" do
+      it "shows correct landing page version" do
+        expect_string("http://#{domain}", "<title>Title version 1</title>")
+
+        release_version(@community.id, 2)
+
+        expect_string("http://#{domain}", "<title>Title version 2</title>")
+      end
+    end
+
+    describe "caching" do
+      before do
+        APP_CONFIG.clp_cache_time = 60
```

Comment:
Shouldn't you save the original `clp_cache_time` and restore that in `after` block instead of hard-coding the 900 seconds, which may change?